CourseSearchAPIHttpTrigger/tests/test_query.py

```diff
@@ -43,3 +43,72 @@ def test_when_both_options_selected(self):
         self.query_params['distance_learning'] = True
         expected = f'({self.doc} eq 0 or {self.doc} eq 1 or {self.doc} eq 2)'
         self.execute(expected)
+
+
+class TestCampusDistanceWithCountry(unittest.TestCase):
+
+#  -----------------------------------------------------------------------------------------------------------------------------------------------------------
+# | On campus | Distance learning |   Wales   |   Scotland   |    N.I.   | KISCOURSE/DISTANCE value | Alternative | Expected no of courses returned (approx.) |
+# |   :---:   |      :---:        |   :---:   |    :---:     |   :---:   |           :---:          |     :---:   |                 ---                       |
+# |     O     |        X          |     O     |      -       |     -     |         1 or 2           |     ne 0    | 149 courses from 9 providers              |
+# |     -     |        -          |     -     |      O       |     O     |         1 or 2           |     ne 0    | 546 courses from 12 providers             |
+# |     O     |        X          |     -     |      O       |     O     |         1 or 2           |     ne 0    | 546 courses from 12 providers             |
+#  -----------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    doc = 'course/distance_learning/code'
+
+    def setUp(self):
+        self.query_params = {}
+        self.filters = []
+    
+    def execute(self, expected):
+        actual = Query.build_or_distance_filter(self.query_params, self.filters)
+        self.assertEqual(actual, expected)
+
+
+    # Test to see if on_campus is removed from the array
+    def test_when_distance_and_campus_selected(self):
```

Comment:
Does it "Test to see if on_campus is removed from the array" or "test_when_distance_and_campus_selected" ?

Consider renaming the test to better explain what it really does then there's no need to add a comment - same for the other tests.

#selfDocumentingCode

Reply:
Renamed the tests and removed the comments